provider/maas/add-juju-bridge.py

```diff
@@ -418,6 +425,16 @@ def main(args):
         print_stanzas(stanzas)
         exit(0)
 
+    cur = StringIO()
+    new = StringIO()
+
+    print_stanzas(stanzas, new)
+    print_stanzas(parser.stanzas(), cur)
+
+    if cur.getvalue() == new.getvalue():
```

Comment:
Given you just did all the work to generate the stanzas, shouldn't we just use that and copy the new.getvalue() to the output, instead of calling print_stanzas() again later?

I also see a 'print_stanzas' call up above, seems like we could avoid doing it multiple times.